### DISCUSSION TITLE
React code uploaded on deployment

### DISCUSSION THREAD

**Discussion by Abd-Ur-Rehman** [vercel/community#2882] [Help]
### Summary

Hello, 

I'm wondering why my app's react code is uploaded/included in the static files. 
<img width="221" alt="image" src="https://github.com/vercel/community/assets/7280837/4944c769-9cf1-4856-a29f-c082ac0b44a8">

### Steps to Reproduce

Go to https://create-react-template.vercel.app/ and check the source files and you'll see those files.

**Comment by amyegan** [comment]
Hi @Abd-Ur-Rehman. I don't have enough context to be able to say exactly why this is happening. Do you have a public repo or [minimal reproduction](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that you can share?

Short of a full repo, it would be helpful to see a screenshot of your repo file directory and a walkthrough of your deployment steps 🙏

**Reply by Abd-Ur-Rehman** [reply]
Here's the GitHub repo. https://github.com/Abd-Ur-Rehman/create-react-app

Deployed here: https://create-react-app-theta-taupe-38.vercel.app/

So if you run the `npm run build`, it only creates the main bundled js file so, why does Vercel have all the source code included on deployed version that you can on deployed URL? 

It must not include these ones. 
<img width="358" alt="image" src="https://github.com/vercel/community/assets/7280837/2144e724-2de6-4231-a197-bb1370050427">

...as they're not when ran the `npm run build` locally. 

```sh
build
├── asset-manifest.json
├── favicon.ico
├── index.html
├── logo192.png
├── logo512.png
├── manifest.json
├── robots.txt
└── static
    ├── css
    │   ├── main.073c9b0a.css
    │   └── main.073c9b0a.css.map
    ├── js
    │   ├── 496.cf873d78.chunk.js
    │   ├── 496.cf873d78.chunk.js.map
    │   ├── main.9ea756af.js
    │   ├── main.9ea756af.js.LICENSE.txt
    │   └── main.9ea756af.js.map
    └── media
        └── logo.6ce24c58023cc2f8fd88fe9d219db6c6.svg
```
So on Vercel, it has only to have the tree's files. 
<img width="380" alt="image" src="https://github.com/vercel/community/assets/7280837/87b9d052-4cf8-4cf8-9190-aa8de63e2f44">

**Reply by Abd-Ur-Rehman** [reply]
@amyegan I hope that you were able to reproduce it. Why it's doing that? Or am I missing any settings in the Vercel project settings? Thoughts, please?

**Reply by Abd-Ur-Rehman** [reply]
@amyegan Just wondering, were you able to reproduce it?

**Reply by amyegan** [reply]
Sorry for the delay, @Abd-Ur-Rehman. I think I understand what you're saying, but this looks like normal behavior for a create-react-app site. You can see the same thing with my own CRA test app: https://create-react-app-vitals.vercel.app/

**Reply by Abd-Ur-Rehman** [reply]
Thanks for getting back @amyegan. Well, it's not of CRA but only happening on Vercel. If you create the app with CRA in your Mac/Windows and then run the `npm run build` which also does the Vercel. It doesn't have those files. See below. 

<img width="301" alt="image" src="https://github.com/vercel/community/assets/7280837/134e8690-06c2-4a52-b72d-895ace751ab2">

So why on Vercel with the same command it has those added in the build js folder? Or let me know if I'm missing any settings on Vercel which could prevent this and build it as it does locally.

**Reply by fabiancook** [reply]
This appears to be the same as the problems I was having here: https://github.com/orgs/vercel/discussions/2737#discussioncomment-6181009 

I had files that were going to be imported by the client, and aren't used by default by the server side build, so for whatever reason are excluded by the vercel build process. 

Adding a direct reference to the files... made it work. This might not work for you...

**Reply by Abd-Ur-Rehman** [reply]
I'm not sure if this is related. How could I add the reference to prevent it? I found a thread on StackOverflow and there wasn't any helpful answer. https://stackoverflow.com/questions/61046858/reactjs-exposing-normal-folder-structure-even-after-build
